python/tests-integration/test_signer.py

```diff
@@ -0,0 +1,140 @@
+import os
+import pathlib
+import subprocess
+import tempfile
+
+import pytest
+
+DATA_DIRECTORY = pathlib.Path(__file__).parents[2].absolute() / "data"
+
+_GPG_INSTRUCTIONS = """
+Key-Type: EDDSA
+  Key-Curve: ed25519
+Subkey-Type: ECDH
+  Subkey-Curve: cv25519
+Name-Real: Integration test key
+Expire-Date: 0
+%no-protection
+%commit
+"""
+
+
+@pytest.fixture
+def ephemeral_gpg_keys() -> None:
+    """Generate public and private GPG keys.
+
+    :yields: Tuple of paths to private and public GPG key.
+    """
+    dir = tempfile.TemporaryDirectory()
+
+    # Create fresh GPG keys
+    with open(f"{dir.name}/instructions", "w") as f:
+        f.write(_GPG_INSTRUCTIONS)
+    genkey = subprocess.run(
+        [
+            "gpg",
+            "--batch",
+            "--generate-key",
+            "--pinentry-mode",
+            "loopback",
+            f"{dir.name}/instructions",
+        ],
+        input=os.devnull,
+        capture_output=True,
+        check=False,
+        text=True,
+        env={"GNUPGHOME": dir.name},
+    )
+    if genkey.returncode != 0:
+        print(genkey.stderr)
+        raise pytest.fail()
```

Comment:
that's a big ugly TBH :)
there are already ways to log and raise cleanly for this kind of situations, so my suggestion is:
```python
print(genkey)
genkey.check_returncode()
```
ie unconditionally print the result of the process, so will be logged by pytest, and then raise in case of failure

Reply:
here `check_returncode()` was not applied, unlike done elsewhere, it seems

Reply:
:arrow_up: